submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/helpers/SourceGenerator.kt

```diff
@@ -20,50 +23,53 @@ class SourceGenerator(
     fun userSources(
         user: SecurityUser,
         rootPath: String? = null
-    ): FilesSource = ComposedFileSource(userSourcesList(user, rootPath.orEmpty()))
+    ): FilesSource = ComposedFileSource(userSourcesList(user, rootPath.orEmpty()), USER_SPACE)
 
     fun submissionSources(requestSources: RequestSources): FilesSource {
-        val (owner, submitter, files, rootPath, submission) = requestSources
-        return ComposedFileSource(submissionSources(owner, submitter, files, rootPath, submission))
+        val (owner, submitter, files, rootPath, submission, preferredSource) = requestSources
+
+        val submissionSources = buildList {
+            add(FilesListSource(files, USER_SPACE))
+            addUserSources(submitter, owner, rootPath)
+            addSubmissionSources(submission)
+        }
+
+        return ComposedFileSource(submissionSources, preferredSource)
     }
 
-    private fun submissionSources(
-        owner: SecurityUser?,
+    private fun MutableList<FilesSource>.addUserSources(
```

Comment:
I am not sure about this, IMO it does not look correct (semantically not sintaxtically)that a mutable list has a business operation `addUserSources` I mean is a list which usually has operation as `elementAt(index)` or `addElement`.